test-runner/src/main/scala/com/typesafe/spark/test/mesos/mesosstate/StateDump.scala

```diff
@@ -37,8 +39,18 @@ object MesosCluster {
     apply(ConfigFactory.parseURL(url))
   }
 
-  def loadStates(mesosConsoleUrl: String): MesosCluster = {
-    MesosCluster(new URL(s"${mesosConsoleUrl}/state.json"))
+  def apply(state: String): MesosCluster = {
+    apply(ConfigFactory.parseString(state))
+  }
+
+  def loadStates(mesosConsoleUrl: String, authToken: Option[String] = Option.empty): MesosCluster = {
+    val url = s"${mesosConsoleUrl}/state.json"
+    val state = Http(url)
+      .header("Authorization", s"token=${authToken.get}")
```

Comment:
This fails with  java.util.NoSuchElementException: None.get if not token is available.